[Synthetics] Add docs on encryption and security (#2953)
elastic/observability-docs: docs/en/observability/synthetics-security-encryption.asciidoc

[[synthetics-security-encryption]]
= Synthetics Encryption and Security

Elastic Synthetics was designed with security in mind encrypting both persisted and transmitted data.
This page catalogs the points within the Elastic Synthetics app data is either stored or transmitted in an encrypted fashion.

> == Synthetics UI (Kibana App)

Reviewer: ```suggestion
[discrete]
== Synthetics UI (Kibana App)
```

> Data is stored in Kibana Secure Saved Objects[https://www.elastic.co/guide/en/kibana/master/xpack-security-secure-saved-objects.html], 

Reviewer: ```suggestion
Data is stored in {kibana-ref}/xpack-security-secure-saved-objects.html[Kibana Secure Saved Objects], 
```

> with sensitive fields encrypted. These fields include your script source, params, and global params.

Reviewer 2: This is even more comprehensive than this, but not sure if we need to specify every field.

Author: Yeah, I don't think we need them all

> == Synthetics Service

Reviewer: ```suggestion
[discrete]
== Synthetics Service
```

The Global Elastic Synthetics Service performs all communication of sensitive data (both internally, and with Kibana) over encrypted connections
and encrypts all data persisted to disk as well.

> == Synthetics Private Locations

Reviewer: ```suggestion
[discrete]
== Synthetics Private Locations
```

> Data is stored in Kibana Secure Saved Objects[https://www.elastic.co/guide/en/kibana/master/xpack-security-secure-saved-objects.html], and encrypted

Reviewer 2: I don't know about this one. Does `data` here refer to the integration policies themselves? Sensitive fields otherwise encrypted in our Synthetics UI are stored as plaintext in policies.

Reviewer 2: Yeah just checked. Integration policies are regular saved objects.

Author: Hmmm, so any encryption provided would be at the OS level (or CSP infra level), and of course there would be the perms on the object. How restrictive are those?

Reviewer 2: You need `Integrations: Read` and `Fleet: All` permissions to read integration policies.

Reviewer 3: We will be adding the ability to encrypt fields on Agent policies and preventing retrieval from Kibana in the near-future: https://github.com/elastic/ingest-dev/issues/1000

> Data is stored in Kibana Secure Saved Objects[https://www.elastic.co/guide/en/kibana/master/xpack-security-secure-saved-objects.html], and encrypted

Reviewer: ```suggestion
Data is stored in {kibana-ref}/xpack-security-secure-saved-objects.html[Kibana Secure Saved Objects], and encrypted
```

> in transit. See https://www.elastic.co/guide/en/fleet/current/_elastic_agent_configuration_encryption.html for more details.

Reviewer: ```suggestion
in transit. See {fleet-guide}/_elastic_agent_configuration_encryption.html[Elastic Agent configuration encryption] for more details.
```
